### PR TITLE
Inclusão do CNAB do Santander

### PR DESCRIPTION
**Pull request by lepri**

### CODE REVIEW

**Comment by mileo** [comment]
@lepri Muito bom! Vou tentar revisar ainda essa semana!

Obrigado

File: l10n_br_account_banking_payment_cnab/febraban/cnab_240/cnab_240.py [hunk 1]
```diff
@@ -49,9 +52,23 @@ def get_bank(bank):
         elif bank == '104':
             from bancos.cef import Cef240
             return Cef240
+        elif bank == '033':
+            from bancos.santander import Santander240
+            return Santander240
         else:
```
Comment (leorochael) [inline l10n_br_account_banking_payment_cnab/febraban/cnab_240/cnab_240.py:58]: This sequence of `if ... elif` clauses should be replaced by a dictionary defined at the module level, e.g.:
    BANK_MAP = {
        '341': bancos.itau.Itau240,
        '104': bancos.bradesco.Bradesco240,
        ...
    }

Even better would be to put the bank number as an attribute each `Cnab240` subclass, and make the `Cnab240` class belong to a metaclass that populates this dictionary automatically when a subclass of `Cnab240` is generated.

File: l10n_br_account_banking_payment_cnab/febraban/cnab_240/cnab_240.py [hunk 2]
```diff
@@ -49,9 +52,23 @@ def get_bank(bank):
...
+    def get_cnab_bank(bank):
+        if bank == '341':
+            from cnab240.bancos import itau
+            return itau
+        elif bank == '033':
+            from cnab240.bancos import santander
+            return santander
+        else:
+            raise NotImplementedError
+
```
Comment (leorochael) [inline l10n_br_account_banking_payment_cnab/febraban/cnab_240/cnab_240.py:71]: This sequence of `if ... elif` should also be replaced by a module level dictionary.

Ideally, this dictionary should belong to the `cnab240` module itself, but until there is such a dictionary there, a dictionary the level of this module would be better than this list of `if ... elif`

File: l10n_br_account_banking_payment_cnab/febraban/cnab_240/cnab_240.py [hunk 3]
```diff
@@ -163,9 +180,62 @@ def remessa(self, order):
...
+        evento = Evento(self.bank, codigo_evento)
+
+        count = 0
         for line in order.line_ids:
-            self.arquivo.incluir_cobranca(**self._prepare_segmento(line))
+
+            seg = self._prepare_segmento(line)
+            if int(banco_num) == 33:
+                seg_s = banco.registros.SegmentoS(**seg)
+            if seg_s.mensagem_recibo_pagador == 2 and count == 0:
```
Comment (leorochael) [inline l10n_br_account_banking_payment_cnab/febraban/cnab_240/cnab_240.py:195]: if the line `if int(banco_num) == 33` is false, then the `seg_s` variable is false and this line will raise `UnbountLocalError`.

File: l10n_br_account_banking_payment_cnab/febraban/cnab_240/cnab_240.py [hunk 4]
```diff
@@ -163,9 +180,62 @@ def remessa(self, order):
...
+            evento.adicionar_segmento(seg_p)
+
+            seg_q = banco.registros.SegmentoQ(**seg)
+            evento.adicionar_segmento(seg_q)
+
+            seg_r = banco.registros.SegmentoR(**seg)
+            if seg_r.necessario():
+                evento.adicionar_segmento(seg_r)
+
+            if (banco_num) == 33:
```
Comment (leorochael) [inline l10n_br_account_banking_payment_cnab/febraban/cnab_240/cnab_240.py:209]: the previous check with `banco_num` was `if int(banco_num) == 33`.

Here the test is just `if (banco_num) == 33`, with superfluous parenthesis.

Is `banco_num` an int or a string?

**Comment by mileo** [comment]
Please send your pr to original repo

**Pull request closed without merging**